src/scalarstats.jl

```diff
@@ -146,6 +162,28 @@ function modes{T}(a::AbstractArray{T})
     return ms
 end
 
+"""
+    mode{T}(a::AbstractArray{T}, rgn::UnitRange{T})
+
+### Args:
+* `a`: An arbitrary array of type `AbstractArray`.
+* `rgn`: A range of integer values, which is optional.
+
+Compute the first mode of `a`, over the range `rng` if specified.
```

Comment:
"rgn". By the way, it would be cool to rename this argument to `r` both in the docs and in the code, to avoid the confusion with RNG.